text/0039-service-overhaul.md

```diff
@@ -0,0 +1,462 @@
+
+Service plugin like integrations one should have a JSON root that look like this:
+```json
+{
+    "format_version": 4,
+    "service": {}
+}
+```
+`"service"` is the service object itself.
+
+##### Plugins specific additions
+Those additions are added like extensions to the service object format.
+
+Getters for those customisations are implemented in the plugin itself.
+- In `obs-services`:
+  - `"id"` (required): service identifier, services are no longer identified by their name. Added because this plugin is multi-services.
+  - `"name"` (required): name of the service. Added because this plugin is multi-services.
+  - `"common"` (optional, false by default): this object is added in service objects of this plugin to identify which service in "common" or not. No third-party plugin should have this behavior.
+- In `obs-youtube`:
+  - Servers name are translated.
+  - `"name_suffix"` in `"servers"` (optional): suffix added to the server name after being translated.
+
+Specific documentations about adding service in `obs-services` should be made.
+
+##### Service object format
+<sup>S</sup>: Should be required by any first-party plugin
+
+<sup>O</sup>: Optional in `obs-services`, can be required or not used at all in first-party one-service plugins depending on their implementations.
+
+- `"more_info_link"`<sup>O</sup>: link with more info about the service.
+- `"stream_key_link"`<sup>O</sup>: link where the user can find his stream key.
+- `"servers"`<sup>S</sup>: array of servers.
+  - `"name"` (required): name of the server.
+  - `"protocol"` (required if if the url prefix can't help identify the protocol, like HLS protocol): protocol identifier (e.g. RTMP, SRT, RIST).
+  - `"url"` (required): url of the server.
+- `"supported_codecs"`<sup>O</sup>: codec not supported by a protocol will not be used. If not set, fallback to what the protocol can support.
+  - `"video"`<sup>O</sup>: video codecs supported by the service.
+    - `"any_protocol"`<sup>O</sup>: array of strings with codecs' name supported by the service for any protocol.
+    - *`"protocol name"`*<sup>O</sup>: (e.g. `"HLS":`) array of strings with codecs' name supported by only this protocol. This array is merged with `"any_protocol"`.
+  - `"audio"`: audio codecs supported by the service.
+    - `"any_protocol"`<sup>O</sup>: array of strings with codecs' name supported by the service for any protocol.
+    - *`"protocol name"`*<sup>O</sup>: (e.g. `"HLS":`) array of strings with codecs' name only supported by this protocol. This array is merged with `"any_protocol"`.
+- `"supported_resolutions"`<sup>O</sup>: array of strings with resolutions (e.g. `"1280x720"`) supported by the service for any protocol.
+  - `"with_framerates"`<sup>O</sup>: default to false. If false each entry shall be put as this "1920x1080", if true it will be put with a framerate as this "1920x1080@30". Only those framerate will be be considered supported.
+  - `"array"`(required): Array of resolutions
```

Comment:
`"with_framerates"` should not be necessary. A basic regular expression (or even a contains check) would easily determine if a frame rate is included. This way `"supported_resolutions"` can just be an array, rather than an object with an `"array"` key. The simpler the JSON structure, the better.

Reply:
Will be removed thanks to JSON Schema conditional pattern matching that work exactly like needed to not need `"with_framerates"`.